rfc/2022/2022-06-16_platform-leadership_002_github-access-managed-nested-teams.md

```diff
@@ -0,0 +1,126 @@
+
+This makes management of users and permissions difficult for multiple reasons:
+1. The roles of various individual users added to the `vets.gov-write` team are unclear and difficult to track.
+1. Onboarding and offboarding of individual users has to be performed by maintainers of `vets.gov-write` creating maintainence burden.
+1. Refined permissions cannot be applied to various level groupings such as "all Platform teams" or "all VFS teams"
+
+We need to restructure our Github Teams to improve clarity, reduce manual overhead, and increase granularity of permissions.
+
+
+## Motivation
+
+Which problem(s) does this solve? Why? Use active voice, present tense, and decisive language.
+Example: Framework XYZ is a modern, well-known, web rendering framework. It is easy to learn, easy to use, and provides high-performance web rendering. etc
+
+## Design
+
+Explain the proposed design in enough detail so that a team member will fully understand the implementation. Include a diagram (in the `images` dir) as needed to convey your plans. Use active voice, present tense, and decisive language.
+
+### Phase 1 - Establish structure
+
+First, use the existing `vets.gov-write` team and ensure all users are represented in a child team.
+
+```mermaid
+graph TD
+    subgraph vets.gov-write
+        subgraph va.gov-platform-teams
+            platform-leadership
+            platform-governance
+            platform-infrastructure
+            platform-release-tools
+        end
+        subgraph va.gov-vfs-teams
+            vfs-example-a
+            vfs-example-b
+            vfs-example-c
+        end
+    end
+```
+
+### Phase 2 - Migrate permissions
+
+- Review [`vets.gov-write` permissions](https://github.com/orgs/department-of-veterans-affairs/teams/vets-gov-write/repositories)
+- Add permissions to one of:
+    - `va.gov-all-teams`
+    - `va.gov-platform-teams`
+    - `va.gov-vfs-teams`
+- Examples:
+    - `Write` access to `va.gov-team`
+    This repo is used by both Platform and VFS teams, so the permission should be set to `va.gov-all-teams`
+    - `Write` access to `platform-console-api`
+    This repo is used only by Platform teams, so the permission should be set to `va.gov-platform-teams`
+
+
+### Phase 3
+
+Move newly created teams `va.gov-platform-teams` and `va.gov-vfs-teams` out of `vets.gov-write` and into `va.gov-all-teams`
+
+```mermaid
+graph TD
+    subgraph va.gov-all-teams
+        subgraph va.gov-platform-teams
+            platform-leadership
+            platform-governance
+            platform-infrastructure
+            platform-release-tools
+        end
+        subgraph va.gov-vfs-teams
+            vfs-example-a
+            vfs-example-b
+            vfs-example-c
+        end
+    end
+```
+
+
+
+
+
+## Risks
```

Comment:
What happens when people swap teams but don't go back through orientation? 
Does this create more work for the Governance team? What does that process look like? Should platform really be the ones setting up these teams and granting permission when we have to rely heavily on the teams to update us on changes in personnel because we aren't privvy to contract changes? I would suggest the OCTODE leads are the owners/maintainers of the teams and they are responsible for maintaining the team because they should get updates about people rolling off or rolling on.